common/test/TestDevice.cs

```diff
@@ -201,7 +201,7 @@ public DeviceClient CreateDeviceClient(ITransportSettings[] transportSettings, C
             return deviceClient;
         }
 
-        private static string GetHostName(string iotHubConnectionString)
+        public static string GetHostName(string iotHubConnectionString)
```

Comment:
This change may not be required. Follow up comments in DeviceClientX509AuthenticationE2ETests.cs